File: src/gtfs/models/Calendar.py
from pydantic import BaseModel, ConfigDict


class Calendar(BaseModel):
    """GTFS calendar entry."""
    model_config = ConfigDict(frozen=True)

    service_id: str
    monday: bool
    tuesday: bool
    wednesday: bool
    thursday: bool
    friday: bool
    saturday: bool
    sunday: bool
    start_date: str  # YYYYMMDD
    end_date: str  # YYYYMMDD

File: src/main/kotlin/com/raptor/gtfs/models/Calendar.kt
package com.raptor.gtfs.models

import kotlinx.serialization.Serializable

@Serializable
data class Calendar(
    val service_id: String,
    val monday: Boolean,
    val tuesday: Boolean,
    val wednesday: Boolean,
    val thursday: Boolean,
    val friday: Boolean,
    val saturday: Boolean,
    val sunday: Boolean,
    val start_date: String,
    val end_date: String
)
